libraries/c_sdk/standard/mqtt/src/iot_mqtt_api.c

```diff
@@ -525,13 +505,28 @@ static void _destroyMqttConnection( _mqttConnection_t * pMqttConnection )
     IotMqtt_Assert( pMqttConnection->pingreqPacketSize == 0 );
 
     /* Remove all subscriptions. */
-    IotMutex_Lock( &( pMqttConnection->subscriptionMutex ) );
-    IotListDouble_RemoveAllMatches( &( pMqttConnection->subscriptionList ),
-                                    _mqttSubscription_setUnsubscribe,
-                                    NULL,
-                                    _mqttSubscription_tryDestroy,
-                                    offsetof( _mqttSubscription_t, link ) );
-    IotMutex_Unlock( &( pMqttConnection->subscriptionMutex ) );
+    if( IotMutex_Take( &( connToContext[ contextIndex ].subscriptionMutex ) ) == true )
+    {
+        if( IotMqtt_RemoveAllMatches( ( connToContext[ contextIndex ].subscriptionArray ), NULL ) == false )
+        {
+            subscriptionStatus = false;
+        }
+
+        IotMqtt_Assert( subscriptionStatus == true );
```

Comment:
please comment why this needs to be asserted.

Reply:
Addressed